spyder/app/tests/test_mainwindow.py

```diff
@@ -6188,5 +6233,58 @@ def test_clickable_ipython_tracebacks(main_window, qtbot, tmpdir):
     qtbot.wait(500)
 
 
+def test_recursive_debug_exception(main_window, qtbot):
+    """
+    Test That an exception in a recursive debug does not break the debugger.
+    """
+    # Wait until the window is fully up
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(
+        lambda: shell.spyder_kernel_ready and shell._prompt_html is not None,
+        timeout=SHELL_TIMEOUT)
+
+    # Main variables
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    control = shell._control
+
+
+    # create new file
+    main_window.editor.new()
+    code_editor = main_window.editor.get_focus_widget()
+    code = 'print("res", 1 + 2)\nprint("res", 2 + 4)'
+    code_editor.set_text(code)
+
+    # debug line
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("debugfile()")
+
+    assert shell.is_debugging()
+    assert '----> 1 print("res", 1 + 2)' in control.toPlainText()
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("debug 1/0")
+
+    assert "ENTERING RECURSIVE DEBUGGER" in control.toPlainText()
```

Comment:
```suggestion
    assert "Entering recursive debugger" in control.toPlainText()
```

To go with my suggestion in https://github.com/spyder-ide/spyder-kernels/pull/286.